backend/README.md

```diff
@@ -45,19 +49,44 @@ LC_ALL=C diesel migration run
 cargo install typeshare-cli
 ```
 
-6. Start the server:
+6. Start Keycloak
 
-```shell
+`docker run -p 8081:8080 -e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD=admin quay.io/keycloak/keycloak:21.1.1 start-dev`
+
+7. Setup Keycloak:
+
+Open <http://localhost:8081/admin>.  
+Sign in with user `admin` and password `admin`.  
+Click on `master` -> `Create Realm`.  
+Name the realm `PermaplanT` and click `Create`.
+
+Click on `Clients` -> `Create client`.  
+Set the `Client ID` to `PermaplanT`.
+Click `Next` two times.  
+Set values: `Root URL = http://localhost:5173`, `Valid redirect URIs = /*`, `Web origins = +`.  
+Click `Save`.
+
+Create a second client `swagger-ui` with `Root URL = http://localhost:8080/doc/api/swagger/ui` (everything else the same as above).
+
+Go to `Users` and create a user `test`.  
+Click `Credentials` and set password to `test`.
+
+8. Run the backend
+
+```bash
 cargo run
 ```
 
-## Usage
+### Test server using Swagger
 
-Now the server is running and will start listening at <http://localhost:8080/> (or whichever port you specified in `.env`).
+Go to <http://localhost:8080/doc/api/swagger/ui/>.  
+Try to execute a request (it should return error 401).  
+Click `Authorize`.
 
-Example requests:
-
-- `curl localhost:8080/api/plants`
```

Comment:
Please also describe how to test via curl.

Reply:
With the setup as described here it is not possible to log in with curl. You would first need to get the access token, but this would require a redirect to Keycloak with a browser based login which is not possible with curl. It would be possible but kind of hacky so I wouldn't suggest it.

Another way of testing would be with Postman, but that is a separate tools so I don't think it fits in the doc.

Reply:
Ok, let us forget about curl but we need some solution for our benchmarking/testing tools. So maybe start describing how to do it with httperf

Reply:
What I did for httperf is I logged in using Postman and copied the token over. Tokens are valid for 5min so benchmarks can only really run for that long. You can also copy the token from the network tab in the browser after you logged in.

Another way to handle that is to use something called client credentials flow. That way you don't actually need a user that is logged in but instead you can fetch a token from keycloak using the client_id and client_secret pair. That is simpler to do with curl.
However you still have to get a new token every 5min so you have to get a token, run httperf for ~4:30min, fetch a new token, ...
You should also use a separate client for that as the client_secret gets in the way of the normal client.

If you think we need to run benchmarks for more than 5min I can add that somewhere in the doc using curl (but it also involves creating a new client).

Reply:
Is there really no way to get tokens valid for longer?

Can you please add what you wrote here in the documentation. Maybe together with how to run benchmarks for the backend.

Reply:
I will do that. I would however do that in a separate PR and merge the currently open frontend one first.
Documentation is listed as a TODO in the auth tracking issue: #329